File: frontend/src/services/recorrenciaService.ts
import api from './api';
import { Recorrencia } from '../types/Recorrencia';

export const recorrenciaService = {
    listarTodas: async (): Promise<Recorrencia[]> => {
        const response = await api.get('/recorrencias');
        return response.data;
    },

    buscarPorId: async (id: number): Promise<Recorrencia> => {
        const response = await api.get(`/recorrencias/${id}`);
        return response.data;
    },

    listarProximas: async (dias: number = 30): Promise<Recorrencia[]> => {
        const response = await api.get(`/recorrencias/proximas?dias=${dias}`);
        return response.data;
    },

    criar: async (data: Omit<Recorrencia, 'id'>): Promise<Recorrencia> => {
        const response = await api.post('/recorrencias', data);
        return response.data;
    },

    atualizar: async (id: number, data: Partial<Recorrencia>): Promise<Recorrencia> => {
        const response = await api.put(`/recorrencias/${id}`, data);
        return response.data;
    },

    toggleAtivo: async (id: number): Promise<void> => {
        await api.patch(`/recorrencias/${id}/toggle`);
    },

    deletar: async (id: number): Promise<void> => {
        await api.delete(`/recorrencias/${id}`);
    },
};

File: backend/src/main/java/com/financeiro/service/RecorrenciaService.java
package com.financeiro.service;

import com.financeiro.dto.RecorrenciaRequestDTO;
import com.financeiro.dto.RecorrenciaResponseDTO;

import java.util.List;

public interface RecorrenciaService {

    List<RecorrenciaResponseDTO> listarTodas();

    RecorrenciaResponseDTO buscarPorId(Long id);

    RecorrenciaResponseDTO criar(RecorrenciaRequestDTO dto);

    RecorrenciaResponseDTO atualizar(Long id, RecorrenciaRequestDTO dto);

    void toggleAtivo(Long id);

    void deletar(Long id);

    List<RecorrenciaResponseDTO> listarProximas(Integer dias);
}
